realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -145,6 +145,39 @@ private SyncConfiguration(File directory,
         this.waitForInitialData = waitForInitialData;
     }
 
+    /**
+     * Returns a SyncConfiguration appropriate to open a synced Realm offline.
+     * This is useful when trying to open a backup Realm (after a Client Reset)
+     *
+     * @param canonicalPath the absolute path to the Realm file defined by this configuration.
+     * @param encryptionKey the key used to encrypt/decrypt the Realm file.
+     * @param modules if specified it will restricts Realm schema to the provided module.
+     * @return SyncConfiguration that can be used offline
+     */
+    public static RealmConfiguration forOffline(String canonicalPath, @Nullable byte[] encryptionKey, Object... modules) {
+        HashSet<Object> validatedModules = new HashSet<>();
+        if (modules.length > 0) {
+            for (Object module : modules) {
+                if (!module.getClass().isAnnotationPresent(RealmModule.class)) {
+                    throw new IllegalArgumentException(module.getClass().getCanonicalName() + " is not a RealmModule. " +
+                            "Add @RealmModule to the class definition.");
+                }
+                validatedModules.add(module);
+            }
+        } else {
+            if (Realm.getDefaultModule() != null) {
+                validatedModules.add(Realm.getDefaultModule());
+            }
+        }
+
+        RealmProxyMediator schemaMediator = createSchemaMediator(validatedModules, Collections.<Class<? extends RealmModel>>emptySet());
+        return forOffline(canonicalPath, encryptionKey, schemaMediator);
+    }
+
+    static RealmConfiguration forOffline(String canonicalPath, byte[] encryptionKey, RealmProxyMediator schemaMediator) {
+        return new RealmConfiguration(null,null, canonicalPath,null, encryptionKey, 0,null, false, OsRealmConfig.Durability.FULL, schemaMediator, null, null, false, null, true);
```

Comment:
should we open it with read-only mode?

Reply:
Would probably be a good idea. I don't see any use cases where writing to it would make sense

Reply:
It would also mean we could remove the modules parameter

Reply:
As discussed this morning we should keep the module option (since not related to the Durability option), we will not restrict it to read only as well, the write obviously will not be synchronised, it's a local Realm after all, the user may which to update some state locally (example delete all rows that have been re-integrated into the new Synced Realm ...)